zenith_utils/src/zid.rs

```diff
@@ -78,6 +78,10 @@ macro_rules! zid_newtype {
             pub fn generate() -> Self {
                 $t(ZId::generate())
             }
+
+            pub const fn from_array(b: [u8; 16]) -> Self {
+                $t(ZId(b))
+            }
```

Comment:
I'd like to use the existing `From` but it doesn't take `const`?

Reply:
Possibly worth making this `#[cfg(test)]`, but maybe there's use cases outside of that.